x-pack/plugins/security_solution/public/plugin.tsx

```diff
@@ -217,6 +225,65 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
     this.services.stop();
   }
 
+  public async registerDiscoverSharedFeatures(
+    core: CoreSetup<StartPluginsDependencies, PluginStart>,
+    plugins: SetupPlugins
+  ) {
+    const { discoverShared } = plugins;
+    const discoverFeatureRegistry = discoverShared.features.registry;
+    const cellRendererFeature: SecuritySolutionCellRendererFeature = {
+      id: 'security-solution-cell-renderer',
+      getRenderer: async () => {
+        const { getCellRendererForGivenRecord } = await this.getLazyDiscoverSharedDeps();
+        return getCellRendererForGivenRecord;
+      },
+    };
+
+    const appWrapperFeature: SecuritySolutionAppWrapperFeature = {
+      id: 'security-solution-app-wrapper',
+      getWrapper: async () => {
+        const [coreStart, startPlugins] = await core.getStartServices();
+
+        const services = await this.services.generateServices(coreStart, startPlugins);
+
+        const { createSecuritySolutionDiscoverAppWrapperGetter } =
+          await this.getLazyDiscoverSharedDeps();
+
+        return createSecuritySolutionDiscoverAppWrapperGetter({
+          core: coreStart,
+          services,
+          plugins: startPlugins,
+        });
+      },
+    };
+
+    const securityReduxStoreInitFeature: SecuritySolutionReduxStoreInitFeature = {
+      id: 'security-solution-redux-store-init',
+      get: async () => {
+        const [coreStart, startPlugins] = await core.getStartServices();
+
+        const subPlugins = await this.startSubPlugins(this.storage, coreStart, startPlugins);
+
+        return this.getStoreForDiscover(coreStart, startPlugins, subPlugins);
+      },
+    };
+
+    discoverFeatureRegistry.register(cellRendererFeature);
+    discoverFeatureRegistry.register(appWrapperFeature);
+    discoverFeatureRegistry.register(securityReduxStoreInitFeature);
```

Comment:
Why do we need to register a feature for the appWrapper and a different feature for the reduxStore? From what I see the store is only used by the app wrapper, so we could register only the appWrapper with the store included in it, isn't it?
I noticed the store is also kept in the `SecurityRootProfileContext`, but I don't see it ever being used, do we foresee some use case where we need to get the store from the profile context?

Reply:
> I noticed the store is also kept in the SecurityRootProfileContext, but I don't see it ever being used, do we foresee some use case where we need to get the store from the profile context?

Store is only kept in `SecurityRootProfileContext` so that it can be passed to `AppWrapper`.  

I initially kept it separate to control the life cycle of store creation vs any other feature which might need it. But you are correct, I can combine `appWrapper` and store initialization. Let me do it rightaway. I do not think, we will need store somewhere else.

Reply:
Done here : df8e005bbbdff7c9bc2a18d5e0bf865e552fbda2